tls/s2n_kem.c

```diff
@@ -23,112 +23,64 @@
 #include "utils/s2n_mem.h"
 #include "utils/s2n_safety.h"
 
-/* If S2N_NO_PQ is set or linked libcrypto doesn't support Kyber, bail on KEM calls.
- * These should never be called.
- */
-int s2n_kyber_kem_keypair_not_supported(IN const struct s2n_kem *kem, OUT uint8_t *pk, OUT uint8_t *sk)
-{
-    POSIX_BAIL(S2N_ERR_UNIMPLEMENTED);
-}
-int s2n_kyber_kem_enc_not_supported(IN const struct s2n_kem *kem, OUT uint8_t *ct, OUT uint8_t *ss,
-        IN const uint8_t *pk)
+#if defined(S2N_NO_PQ)
+int s2n_kyber_evp_generate_keypair(IN const struct s2n_kem *kem, OUT uint8_t *public_key, OUT uint8_t *secret_key)
```

Comment:
Should these unimplemented stubs be implemented in s2n_kyber_evp instead? It looks like there are stubs already implemented there, but with the wrong names maybe? Might be misunderstanding.

https://github.com/aws/s2n-tls/blob/a2b16d2ccf59ed419a4f3043b88563354fc30b5d/pq-crypto/s2n_kyber_evp.c#L91-L94

Reply:
It turns out that the linker couldn't find these in s2n_kyber_evp.c when S2N_NO_PQ=1, which makes sense -- we don't compile pq-crypto/* in that case. So, I think the stubs need to live here.